goinp/goinp.go

```diff
@@ -63,41 +65,39 @@ func AskForString(messageToPrint string) (string, error) {
 }
 
 func writeStdinClearable(text string) error {
-	for _, c := range []byte(text) {
-		if _, _, errno := syscall.Syscall(syscall.SYS_IOCTL, os.Stdin.Fd(), syscall.TIOCSTI, uintptr(unsafe.Pointer(&c))); errno != 0 {
-			return fmt.Errorf("failed to write to stdin, err no: %d", errno)
+	// clearable stdin text is only possible if there is an available terminal input buffer
+	if terminal.IsTerminal(int(os.Stdin.Fd())) {
+		for _, c := range []byte(text) {
+			if _, _, errno := syscall.Syscall(syscall.SYS_IOCTL, os.Stdin.Fd(), syscall.TIOCSTI, uintptr(unsafe.Pointer(&c))); errno != 0 {
+				return fmt.Errorf("failed to write to stdin, err no: %d", errno)
+			}
 		}
 	}
 	return nil
 }
 
-// AskOptions ...
-func AskOptions(title string, defaultValue string, optional bool, options ...string) (string, error) {
+func askOptionsWithReader(title string, defaultValue string, optional bool, reader *bufio.Reader, options ...string) (string, error) {
 	const customValueOptionText = "<custom value>"
 
 	if len(options) == 0 {
```

Comment:
Note: move only the
```if len(options) == 0 {```
part to a function.
Then no recursion is needed and control flow is easier to parse.